### PR TITLE
feat(templates): bring React 18 templates in line with `indie-stack` updates

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Mainly includes https://github.com/remix-run/indie-stack/commit/e8d39cf00d92d2d16abab93025e4b4e47f8c808c, https://github.com/remix-run/indie-stack/commit/a1a1c05dcc2612e20fe35bdff2567df82719b85b & https://github.com/remix-run/indie-stack/pull/125

### CODE REVIEW

**Review by mcansh — APPROVED** [review]
LGTM, but I'll defer to @jacob-ebey

File: templates/express/app/entry.server.tsx
Comment (jacob-ebey) [inline templates/express/app/entry.server.tsx:16]: MJ wants these to be two explicit code-paths, something like:

```js
if (isBot) {
  return renderForBot(...);
}

return renderForBrowser(...);
```

File: templates/fly/app/entry.server.tsx
Comment (jacob-ebey) [inline templates/fly/app/entry.server.tsx:16]: MJ wants these to be two explicit code-paths, something like:

```js
if (isBot) {
  return renderForBot(...);
}

return renderForBrowser(...);
```

File: templates/remix/app/entry.server.tsx
Comment (jacob-ebey) [inline templates/remix/app/entry.server.tsx:16]: MJ wants these to be two explicit code-paths, something like:

```js
if (isBot) {
  return renderForBot(...);
}

return renderForBrowser(...);
```

**Review by jacob-ebey — CHANGES_REQUESTED** [review]

**Review by mcansh — APPROVED** [review]

File: templates/express/app/entry.server.tsx
Comment (mjackson) [inline templates/express/app/entry.server.tsx:5]: This isn't a huge deal, but TS marks this import as missing in dev. I've always tried to avoid this kind of thing in the past. Not sure the best way to fix it here though since this is in our templates.

Maybe we just add `isbot` to our monorepo deps to avoid the TS warning? Just a thought. Open to your suggestions.

Screen Shot 2022-09-29 at 3 49 27 PM

File: templates/arc/app/entry.client.tsx
Comment (mjackson) [inline templates/arc/app/entry.client.tsx:5]: Small nit: we prefer to use `function` for top-level module scope functions.

```ts
function hydrate() {
  // ...
}
```

File: templates/express/app/entry.server.tsx
Comment (mjackson) [inline templates/express/app/entry.server.tsx:43]: Could also use a regular function here to save a few chars:

```suggestion
        onAllReady() {
```

File: templates/express/app/entry.server.tsx
Comment (mjackson) [inline templates/express/app/entry.server.tsx:31]: Please use a regular `function` here to remain consistent with the rest of the file.

File: templates/arc/app/entry.client.tsx
Comment (mjackson) [inline templates/arc/app/entry.client.tsx:6]: 👍  for using this pattern by default in our templates. Very cool 😎

**Review by mjackson — CHANGES_REQUESTED** [review]
Hey @MichaelDeBoey this looks great, thank you. I'd just like a few small syntax changes to remain consistent with the rest of the codebase.

I know I said I could show you how to dedup the code, but I think I prefer it how it is now. It's clear and concise. If people want ternaries, they can make that refactoring themselves.

Thank you! ❤️

File: templates/express/app/entry.server.tsx
Comment (MichaelDeBoey) [reply templates/express/app/entry.server.tsx:5]: #4061 will add `isbot` to the monorepo deps, so these errors should be gone once that one's merged I think.

File: templates/express/app/entry.server.tsx
Comment (MichaelDeBoey) [reply templates/express/app/entry.server.tsx]: These were already arrow functions, so I just kept it like they were.

Will revert

**Review by mcansh — APPROVED** [review]

**Pull request merged**
